sources/oracle/data_test.go

```diff
@@ -0,0 +1,274 @@
+package oracle
+
+import (
+	"math/big"
+	"math/bits"
+	"testing"
+	"time"
+
+	"github.com/cloudspannerecosystem/harbourbridge/internal"
+	"github.com/cloudspannerecosystem/harbourbridge/schema"
+	"github.com/cloudspannerecosystem/harbourbridge/spanner/ddl"
+	"github.com/golang-sql/civil"
+	"github.com/stretchr/testify/assert"
+)
+
+type spannerData struct {
+	table string
+	cols  []string
+	vals  []interface{}
+}
+
+func TestProcessDataRow(t *testing.T) {
+	tableName := "testtable"
+	cols := []string{"a", "b", "c", "d"}
+	conv := buildConv(
+		ddl.CreateTable{
+			Name:     tableName,
+			ColNames: cols,
+			ColDefs: map[string]ddl.ColumnDef{
+				"a": {Name: "a", T: ddl.Type{Name: ddl.Float64}},
+				"b": {Name: "b", T: ddl.Type{Name: ddl.Int64}},
+				"c": {Name: "b", T: ddl.Type{Name: ddl.Timestamp}},
+				"d": {Name: "d", T: ddl.Type{Name: ddl.String, Len: int64(1)}},
+			}},
+		schema.Table{
+			Name:     tableName,
+			ColNames: cols,
+			ColDefs: map[string]schema.Column{
+				"a": {Name: "a", Type: schema.Type{Name: "FLOAT"}},
+				"b": {Name: "b", Type: schema.Type{Name: "NUMER"}},
+				"c": {Name: "c", Type: schema.Type{Name: "TIMESTAMP(6)"}},
+				"d": {Name: "d", Type: schema.Type{Name: "CHAR", Mods: []int64{1}}},
+			}})
+	conv.SetDataMode()
+	var rows []spannerData
+	conv.SetDataSink(func(table string, cols []string, vals []interface{}) {
+		rows = append(rows, spannerData{table: table, cols: cols, vals: vals})
+	})
+	ProcessDataRow(conv, tableName, cols, conv.SrcSchema[tableName], tableName, cols, conv.SpSchema[tableName], []string{"4.2", "6", "2022-01-19T09:34:06.47Z", "p"})
+	assert.Equal(t, []spannerData{{table: tableName, cols: cols, vals: []interface{}{float64(4.2), int64(6), getTime("2022-01-19T09:34:06.47Z"), "p"}}}, rows)
+}
+
+func TestConvertData(t *testing.T) {
+
+	numStr := "33753785954695469456.33333333982343435"
+	numVal := new(big.Rat)
+	numVal.SetString(numStr)
+
+	singleColTests := []struct {
+		name  string
+		ty    ddl.Type
+		srcTy string      // Source DB type
+		in    string      // Input value for conversion.
+		e     interface{} // Expected result.
+	}{
+		{"bytes", ddl.Type{Name: ddl.Bytes, Len: ddl.MaxLength}, "", string([]byte{137, 80}), []byte{0x89, 0x50}}, // need some other approach to testblob type
+		{"date", ddl.Type{Name: ddl.Date}, "", "2019-10-29", getDate("2019-10-29")},
+		{"float", ddl.Type{Name: ddl.Float64}, "", "42.6", float64(42.6)},
+		{"int", ddl.Type{Name: ddl.Int64}, "", "42", int64(42)},
+		{"string", ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, "VARCHAR2", "eh", "eh"},
+		{"number", ddl.Type{Name: ddl.Numeric}, "NUMBER", numStr, numVal},
+		{"timestamp", ddl.Type{Name: ddl.Timestamp}, "TIMESTAMP(6)", "2022-01-19T09:34:06.47Z", getTime("2022-01-19T09:34:06.47Z")},
```

Comment:
Consider making this list exhaustive by adding the remaining types (eg. numeric json etc)

Reply:
covered all spanner types.